java/src/main/java/com/google/appengine/tools/mapreduce/impl/shardedjob/ShardedJobRunner.java

```diff
@@ -85,7 +89,7 @@ private static RetryerBuilder baseRetryerBuilder() {
       .retryIfExceptionOfType(ApiProxyException.class)
       // don't think this is thrown by new datastore lib
       // thrown by us if the task state is from the past
-      .retryIfExceptionOfType(ConcurrentModificationException.class)
+      .retryIfExceptionOfType(IllegalStateException.class)
```

Comment:
changed this, bc I think it more accurate reflects what the situation is.

I don't think it's getting concurrently modified; I think the task is executing before txn is committed OR before txn outcome can be seen